src/dist/component/package.rs

```diff
@@ -163,19 +164,37 @@ struct MemoryBudget {
     used: usize,
 }
 
+static mut RAM_NOTICE_SHOWN: bool = false;
+
 // Probably this should live in diskio but ¯\_(ツ)_/¯
 impl MemoryBudget {
-    fn new(max_file_size: usize) -> Self {
-        const DEFAULT_UNPACK_RAM: usize = 400 * 1024 * 1024;
-        let unpack_ram = if let Ok(budget_str) = env::var("RUSTUP_UNPACK_RAM") {
-            if let Ok(budget) = budget_str.parse::<usize>() {
-                budget
-            } else {
-                DEFAULT_UNPACK_RAM
+    fn new(max_file_size: usize, effective_max_ram: usize) -> Self {
+        const DEFAULT_UNPACK_RAM_MAX: usize = 500 * 1024 * 1024;
+        let ram_for_unpacking = effective_max_ram - (100 * 1024 * 1024);
```

Comment:
Perhaps:
```suggestion
        const RAM_ALLOWANCE_FOR_OS: usize = 100 * 1024 * 1024;
        let ram_for_unpacking = effective_max_ram - RAM_ALLOWANCE_FOR_OS;
```

Reply:
That overhead is for rustup and the tar buffers, but yes, I'll add a constant for clarity.